crypto/fipsmodule/bcm.c

```diff
@@ -278,6 +284,11 @@ int BORINGSSL_integrity_test(void) {
   assert_within(rodata_start, kPrimes, rodata_end);
   assert_within(rodata_start, kP256Params, rodata_end);
   assert_within(rodata_start, kPKCS1SigPrefixes, rodata_end);
+#if defined(OPENSSL_X86) || defined(OPENSSL_X86_64)
+  assert_not_within(rodata_start, OPENSSL_ia32cap_P, rodata_end);
```

Comment:
How can `OPENSSL_ia32cap_P` ever be in the `.rodate` segment? It's not a constant, hence can't be mapped as read-only.

Reply:
On Windows, the regular data segment and read-only data segments are the same: https://github.com/aws/aws-lc/blob/6dcb1c327d4e9023855b92d6f7eb8e2c54e5616f/crypto/fipsmodule/fips_shared_library_marker.c#L38-L39

This combined with the fact that in the 1MU branch, `cpucap.c` was included directly in `bcm.c`, mean that the variable was in the read-only data segment. While this has already been fixed in main (and a fix is coming for the 1MU branch), we want to ensure that if it somehow happens again we catch it early.

Reply:
I'm not sure I get this, on the other hand, my experience with Windows is minimal.

If the equivalent to `.data`/`.bss` in COFF land is lumped together with read-oonly data, how is the hash over it (always) deterministic (with respect to the initial hash computation at compile time) when mapped into memory? We just don't modify any data before the integrity check, apart from `OPENSSL_ia32cap_P`?

Reply:
You're right that this is a problem, but it's also not in the scope of this PR so I created another one that addresses your concerns here: https://github.com/aws/aws-lc/pull/880